Add updated publish documentation section

jayk-u opened the pull request:
This PR adds the following:
- Create and manage documentation page
- Deploy and release management page
- Updated "Branching" page
- Updated "Access management" page

I feel that I still need some help on the technical parts, specifically for CLI/CI/API in "Create and manage doc" + "Deploy and release management".

This is a huge part of our help center, with many important concepts to explain. I'll read carefully your feedbacks :) Thanks!

ChristopheDujarric commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+Public documentation is indexed by search engines and can be found by anyone without any restrictions. It is not possible to restrict access to any part of your documentation; it will be fully accessible.
+
+Private documentation, on the other hand, is only accessible to users who have been individually invited. There are two types of invited users: those who are part of the same organization and "external" guests.
+When documentation is set as private, it is completely private. It is excluded from indexing by search engines, and only authenticated users who know its URL can access it.
```
```suggestion
Private docs require user authentication to be accessible. They are not indexed by search engines.
```

jayk-u replied:
👍

ChristopheDujarric commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+
+Access to your public documentations is solely determined by its URL and its discoverability by search engines.
+
+Your public documentations are accessible at the URL `https://bump.sh/your-slug/your-documentation`.
```
A part le pluriel de documentation, je me demande dans quelle mesure cette phrase invite à une note type "pour utiliser un custom domain, allez voir tel bout de doc"?

scharrier replied:
The URL is wrong. `https://bump.sh/your-slug/doc/your-documentation` (and it can also be under a `/hub`) or a custom domain, yep.

jayk-u replied:
👍

scharrier commented on src/_help/publish-documentation/create-and-manage-documentation.md:
```diff
@@ -0,0 +1,37 @@
+![](/images/help/doc-creation.png)
+
+- Name: The name of the documentation as it will appear to those who have access to it.
+- Hub: Here, you can specify whether you want to link this documentation to an existing hub or not. Hubs are collections of documentation.
+- Slug: The slug allows you to define the URL of your documentation, in the format `https://bump.sh/your-username/your-slug`.
+- Documentation Access: Documentation can be either public and accessible to everyone or private, visible only to you and invited members.
```
Should we use a table here?
Also, can we update the "Create a new hub" page to make it consistent with this one? (in another PR, of course) We don't have this list of attributes on the other page.

scharrier commented on src/_help/publish-documentation/create-and-manage-documentation.md:
```diff
@@ -0,0 +1,37 @@
+- Slug: The slug allows you to define the URL of your documentation, in the format `https://bump.sh/your-username/your-slug`.
+- Documentation Access: Documentation can be either public and accessible to everyone or private, visible only to you and invited members.
+
+The next step is to deploy the first version of your documentation, for which you will need your OpenAPI or AsyncAPI specification file.
```
Please write 💯 times: specification is the standard, API definition is the document ;)

scharrier commented on src/_help/publish-documentation/create-and-manage-documentation.md:
```diff
@@ -0,0 +1,37 @@
+
+![](/images/help/upload-version.png)
+
+There are several methods for deploying the first and future versions of your documentation, which we will detail further in the next section.
```
Please do not use `versions`

scharrier commented on src/_help/publish-documentation/create-and-manage-documentation.md:
```diff
@@ -0,0 +1,37 @@
+
+There are several methods for deploying the first and future versions of your documentation, which we will detail further in the next section.
+
+## Delete a doc
```
Please be consistent with the concepts. We use `documentation`.

jayk-u replied:
Fixed in the last commit

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+## What are deployments
+
+Deployment corresponds to the publication of an uploaded version of documentation.
```
Should we write "The deployment term" or "A deployment"?
Please remove the `version` term here.
We should also define what a release is, here.

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+![](/images/help/deployments-list.png)
+
+By default, a version is automatically deployed after being uploaded. However, you can choose to deploy a version manually at a later time by using the Manual Release mode.
```
Just to add the idea, please rephrase if needed:
```suggestion
By default, a version is automatically deployed after being uploaded. However, you can choose to deploy a version manually at a later time by using the Manual Release mode. In this case, the latest deployments will be enqueued until you decide to release one of them.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+By default, a version is automatically deployed after being uploaded. However, you can choose to deploy a version manually at a later time by using the Manual Release mode.
+
+You can also add a title and description to your last automatic deployment afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.
```
```suggestion
You can also add context (title and description) to any release afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+After processing, the new version will be automatically deployed after a few moments unless you have enabled Manual Release mode.
+
+### Deploy from the CLI
```
I'm in favor of replacing CLI/CI/GitHub action paragraphs with links pointing to the right section.

jayk-u replied:
Last Monday, we decided during the GTM weekly to share this topic to the team as we weren't sure of what should be the most convenient.

To keep it short, [they suggested ](https://itducks.slack.com/archives/C01DP38BRTL/p1699279553854939) to keep a short explanation with a link pointing to te full section, instead of directly pointing to the complete page elsewhere.

As nobody else jumped in the thread, and because we were looking for such feedback to make a decision, I chose to stick to that so far, awaiting for more reviews on the thread and the PR. This is a reason why this part has not moved yet.

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+This option allows you to continue working on your documentation without worrying about the impact on your users. For example, you can prepare a future version of your API and its documentation with members of your organization and launch its publication when the time is right.
+
+#### Release manually a deployment
```
Isn't this better?
```suggestion
#### Manually release a deployment
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+![](/images/help/deployments-list-not-released.png)
+
+The deployment preparation page will appear, allowing you to customize it by adding a title or a brief description, for example, to accompany this new version. These fields support Markdown to make the result more readable.
```
Add a link to our Markdown support.
It's more about adding context than customizing.

scharrier commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+
+## Public or Private?
+
+Bump.sh allows you to choose whether a documentation is public or private.
```
We use "a documentation" here, and then just "documentation"

scharrier commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+
+![](/images/help/documentation-access-choice.png)
+
+## Private
```
I think we should better start with the members of the organization, and then "External guests".

scharrier commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+
+### Invite external guests
+
+To invite people to access a private documentation, click on the `Share Access` button, which is visible in the settings of that documentation.
```
People from outside of your organization.

scharrier commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+
+![](/images/help/share-access.png)
+
+They will receive an email inviting them to sign up on Bump.sh. This step is essential to ensure the security of access to your documentation.
```
Should we add a note about the fact that this isn't possible (yet) on hubs?

jayk-u replied:
Not 100% sure, I think that should be in Hubs section instead?

scharrier commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+
+They will receive an email inviting them to sign up on Bump.sh. This step is essential to ensure the security of access to your documentation.
+
+### Invite members of an organization
+
+Inviting a colleague or teammate to become a member of [your organization](/help/organizations/) can be done from the settings of that organization. Simply enter the email address of the person in question.
+
+![](/images/help/org-add-member.png)
```
It's not about "Inviting", I think this part is just "Members of the organization", and the content should probably just be something like "Any member of your organization can access any doc" or something like that.

scharrier commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,47 @@
+
+Your public documentations are accessible at the URL `https://bump.sh/your-slug/your-documentation`.
+
+> Any changes to your slug/username will result in a change in the URL of your public documentations, without redirection from our side. Make sure to update the URLs you publish or share with your API users accordingly.
```
Or better: use a custom domain ;)

scharrier reviewed (commented):
Great basis to finalize this section 👏 

I did a first review and left (a lot of) comments. Here are also some global feedback:
* remove the "version" term, which is confusing (as it's related to API versions). Replace it with "deployment", "release", or "API definition" according to the context
* the chosen icon for navigation doesn't make sense (it's a paper plane)
* one the hubs pages -> there is not "Where can I find my hubs" section -> consistency
* on the hubs pages -> there is no "delete" section -> consistency
* please always add links when talking to another section
* I don't think we should add the CLI/GitHub action/API help in the feature pages, but point to the related pages.

paulRbr commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+![](/images/help/create-branch.png)
+
+Once you have created your branch, you can start uploading and [deploying your first versions](/help/publish-documentation/deploy-and-release-management/) using your dashboard, the CLI, or the CI.
```
```suggestion
Once you have created your branch, you can start [deploying your first API document](/help/publish-documentation/deploy-and-release-management/) using your dashboard, the CLI, or your CI.
```

jayk-u replied:
👍

paulRbr commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+## Deploy into a branch
+
+Deploying a version in a branch follows the usual procedure, with an additional step where you specify which branch to add this new version to.
```
```suggestion
Deploying an API document in a branch follows the usual procedure, with an additional step where you specify which branch to add this new deploy to.
```

paulRbr commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+Deploying a version in a branch follows the usual procedure, with an additional step where you specify which branch to add this new version to.
+
+From the settings of a documentation, in the "Upload" section, simply select the branch in which to deploy the version before confirming the upload.
```
```suggestion
From the documentation settings, in the "Upload" section, select the branch in which to deploy before confirming the upload.
```

jayk-u replied:
👍

paulRbr commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+### Using the API
+
+Our [API](https://developers.bump.sh/group/endpoint-branches) also allows you to manage the branches of your documentation. You can create a new branch, delete one, list them all, select one as the default branch, and deploy a new version into a branch.
```
```suggestion
Our [API](https://developers.bump.sh/group/endpoint-branches) also allows you to manage the branches of your documentation. You can create a new branch, delete one, list them all, select one as the default branch, and deploy an API document into a branch.
```

jayk-u commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+What we refer to as "Branching" is a solution that allows you to have multiple different versions of the same API documentation on parallel branches, somewhat similar to Git.
+
+Branches can, for example, be used to distinguish between two versions of your API documentation, such as those for production or development environments. Another example is keeping track of previous versions of an API, preserving the history for users who are not using the latest version (e.g., 1.0, 1.1, 2.0, etc.).
```
:+1:

jayk-u commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+Branches can, for example, be used to distinguish between two versions of your API documentation, such as those for production or development environments. Another example is keeping track of previous versions of an API, preserving the history for users who are not using the latest version (e.g., 1.0, 1.1, 2.0, etc.).
+
+You can create as many branches as necessary, and each of these branches can receive its own deploys and has a distinct changelog from others. You can also choose which branch will be the default one displayed.
```
Modified the "default" part. Not the structure from the first comment.

jayk-u commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+## Create a new branch
+
+From the settings of a documentation, the Branches section allows you to create new branches. Simply enter the name of the new branch and confirm.
```
👍

jayk-u commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+Once you have created your branch, you can start uploading and [deploying your first versions](/help/publish-documentation/deploy-and-release-management/) using your dashboard, the CLI, or the CI.
+
+## Deploy into a branch
```
👍

jayk-u commented on src/_help/publish-documentation/create-and-manage-documentation.md:
```diff
@@ -0,0 +1,37 @@
+
+## Where can I find my documentation?
+
+By default, accessing your dashboard displays your personal account, which lists all your documentations.
```
Je suis le premier à avoir milité en arrivant chez Bump.sh sur la correction de "documentations" qu'on mettait souvent au pluriel.
Les 4 `s` en trop dans cette PR sont des erreurs d'étourderie et de fatigue dont je m'excuse platement 🙇

jayk-u commented on src/_help/publish-documentation/create-and-manage-documentation.md:
```diff
@@ -0,0 +1,37 @@
+By default, accessing your dashboard displays your personal account, which lists all your documentations.
+![](/images/help/personal-account.png)
+
+## Create a new documentation
```
👍

jayk-u commented on src/_help/publish-documentation/deploy-and-release-management.md:
````diff
@@ -0,0 +1,125 @@
+Using the CLI, you can also deploy a version using the `deploy` command.
+
+```
+bump deploy path/to/file.json --doc my-documentation
````
👍 
J'ai modifié l'ordre d'apparition de la note aussi au passage.

jayk-u commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+Publishing a new version of your documentation typically follows a two-step process:
+- You upload a new version of your documentation through a specification file.
+- This new version is validated and deployed, becoming the "visible" version of your documentation. The changelog will also be update, reflecting changes between the current and previous version.
```
👍

jayk-u commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,125 @@
+
+### Manual Release
+
+The Manual Release mode was recently introduced on Bump.sh and is still in beta. Please don't hesitate to contact us if you have any questions.
```
👍

jayk-u commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+- TOC
+{:toc}
+
+## What is branching?
```
👍

jayk-u commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,60 @@
+
+## What is branching?
+
+What we refer to as "Branching" is a solution that allows you to have multiple different versions of the same API documentation on parallel branches, somewhat similar to Git.
```
👍

jayk-u commented on src/_help/publish-documentation/branching.md:
````diff
@@ -0,0 +1,60 @@
+
+![](/images/help/upload-branch.png)
+
+### Using the CLI
+
+When deploying via the CLI, it is possible to specify the target branch using the `--branch <branch-name>` parameter. If the branch does not exist, it will be created at this stage.
+
+```
+bump deploy path/to/file.json --doc my-documentation --branch staging
+```
+
+> You will need to pass your private documentation access token for this command to work. Either with the `--token` flag or via the `BUMP_TOKEN` environment variable. This token can be found in your documentation settings under the "CI deployment" page.
+
+{: .info}
+
+### Using the API
+
+Our [API](https://developers.bump.sh/group/endpoint-branches) also allows you to manage the branches of your documentation. You can create a new branch, delete one, list them all, select one as the default branch, and deploy a new version into a branch.
+
+Please note that, currently, the API does not support renaming a branch.
````
I haven't moved that part in the next commit, after the suggestions both from Thimy and Paulr (who shared that when it's a small piece of info, we should keep it). Please let me know what shall we stand for here.

ChristopheDujarric commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,64 @@
+
+## Create a new branch
+
+From the settings of a documentation, the Branches section allows you to create new branches. Enter the name of the new branch and confirm. You can also create branches directly with our CLI or our API.
```
```suggestion
From the documentation settings, the Branches section allows you to create new branches. Enter the name of the new branch and confirm. You can also create branches directly with our CLI or our API.
```

jayk-u commented on src/_data/sidebar.yml:
```diff
@@ -19,10 +19,23 @@ help:
-  - type: page
-    label: Release Management
-    link: /publish-documentation/release-management/
+  - type: category
```
This was not in the initial outline and, I could be wrong here, I don't think we clarified or made a clear decision on having this format as a standard across the whole help center.

I've added the page as requested, and tried to keep the spirit of the other ones. I felt that I could have added a glossary part (specifically the one about the concepts and terminology around deployments (API Def/Doc/Change/etc...)), but as this part is not final yet and obviously need some rework and clarification, I'll wait for the review.

scharrier replied:
Thanks for the clarification. We did have this discussion for Hubs (or organizations), so I thought it was a clear decision to add these "index" pages to all the categories to have consistent behavior across the whole help center, sorry if it wasn't.

jayk-u commented:
I've just pushed another commit for this PR, here's what I've changed:

- added a new index page to the section to keep consistency between the other new sections
- fixed a few typos (thought that some of them were in the last commit, my apologies)
- updated the sidebar (some elements has been deleted or moved)

The introduction/glossary part of the `Deploy and release management` page awaits your review, as I've understood that I missed the target with this first draft.

Finally, after requesting team feedback on Slack Monday, I haven't removed yet the CLI/CI/API parts in these pages. The suggestion made by Thimy and Paulr was to keep very small and specific parts, with a link pointing to the fully detailed section.
In my opinion, I think that could work well (even if can become troublesome to maintain) but I'd like to check with everyone before moving on.

Thanks.

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+## What are deployments
+
+Deployment corresponds to the publication of an uploaded API definition of documentation.
+
+We call an "API document" the file that contains your API definition, the content of your API. When you upload and then deploy an API document, it creates or updates an API version: a marked and named iteration of your API.
+
+When your API undergoes a structural change, it is reflected on Bump.sh in the changelog. This change is referred to as an "API change."
+
+![](/images/help/deployments-list.png)
+
+By default, an API document is automatically deployed after being uploaded. However, you can choose to deploy it manually at a later time by using the Manual Release mode. In this case, the latest deployments will be enqueued until you decide to release one of them.
+
+You can also add context (title, description) to any release afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.
```
I reworked this part to make it easier to understand (and fixed the mistakes). WDYT of this proposal?

```suggestion
A Deployment is the processing of an API document - containing your API definition - sent to Bump.sh servers. 

Following this deployment, the new API definition is published to your API consumers. This is what we call a Release: your API documentation is updated, and if your API has changed since the last deployment, a new entry representing this API change is added to your API changelog.

![](/images/help/deployments-list.png)

By default, a deployment is automatically released after its processing. However, you can choose to control the release flow by activating the Manual Release setting. In this case, the latest deployments will be enqueued until you decide to release one of them.

You can also add context (title, description) to any release afterward. This option is available from the page of the deployment, accessible in the list of deployments for your documentation.
```

(Do not hesitate to fix my English if needed, it's more about simplifying the concepts that working on sentences)

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+### Deploy from the dashboard
+
+To deploy a new API definition of documentation from the dashboard, you first need to go to the settings of that documentation.
```
```suggestion
To deploy a new API document from the dashboard, you first need to go to the settings of your documentation.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+To deploy a new API definition of documentation from the dashboard, you first need to go to the settings of that documentation.
+
+From there, you will find the "Deploy new version" button, which will allow you to choose the file to upload and, if the documentation has branches, the branch to deploy it on.
```
We need to rename this button!

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+![](/images/help/upload-document-quick.png)
+
+After processing, the new API document will be automatically deployed after a few moments unless you have enabled Manual Release mode.
```
```suggestion
After processing, the new API document will be automatically released after a few moments unless you have enabled Manual Release mode.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+### Deploy from the CLI
+
+Our CLI can also help with deploying an API document using the `deploy` command. The complete process is available on the dedicated [CLI page](/help/continuous-integration/index/).
```
```suggestion
You can deploy your API document through our CLI using the `deploy` command. The complete process is available on the dedicated [CLI page](/help/continuous-integration/index/).
```

scharrier replied:
Also, the link is not the good one (it's the CLI page, not the CI one).

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+## Release management
+
+Publishing a new API definition of your documentation typically follows a two-step process:
```
```suggestion
Releasing a new API definition typically follows a two-step process:
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+## Release management
+
+Publishing a new API definition of your documentation typically follows a two-step process:
+- You upload a new API definition of your documentation through an API document, your specification file.
```
```suggestion
- You upload a new API document through your dashboard, the CLI, our GitHub action or our API
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+Publishing a new API definition of your documentation typically follows a two-step process:
+- You upload a new API definition of your documentation through an API document, your specification file.
+- This new API document is validated and deployed, becoming the "visible" API definition of your documentation. The changelog will also be update, reflecting changes between the current and previous definitions.
```
```suggestion
- This new API document is validated and deployed, becoming the "visible" API definition presented in your documentation. The changelog is also updated, reflecting changes between the current and previous definitions.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+- You upload a new API definition of your documentation through an API document, your specification file.
+- This new API document is validated and deployed, becoming the "visible" API definition of your documentation. The changelog will also be update, reflecting changes between the current and previous definitions.
+
+By default, these two steps are automatically linked: uploading a new API document results in its publication. However, it is possible to dissociate them using the Manual Release mode.
```
```suggestion
By default, these two steps are automatically linked: uploading a new API document results in its release. However, it is possible to dissociate them using the Manual Release mode.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+By default, these two steps are automatically linked: uploading a new API document results in its publication. However, it is possible to dissociate them using the Manual Release mode.
+
+The Deployments section of your documentation provides access to the history of all API documents uploaded on Bump.sh, including their status (`enqueued`, `deploying`, `deployed`, or `errored`) and their release status (`Released` or `Not released`).
```
Not sure about this one:
```suggestion
The Deployments section of your documentation provides access to the history of all the API documents uploaded on Bump.sh, including their status (`enqueued`, `deploying`, `deployed`, or `errored`) and their release status (`Released` or `Not released`).
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+The Manual Release mode is still in beta. Please don't hesitate to contact us if you have any questions.
+
+The Manual Release mode is an option that allows you to separate the upload step from the publication step of your API documentation.
```
```suggestion
The Manual Release mode is an option that allows you to separate the processing step from the release step of a new API definition.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+![](/images/help/manual-release-toggle.png)
+
+When activated, uploading an API document of your API documentation will no longer automatically publish it. We continue to prepare this API definition on our side (and you can view a preview and its changelog), but the release step becomes manual. This means you can decide when to make an API definition live for your API users.
```
```suggestion
When activated, uploading an API document will no longer automatically release it. We continue to prepare this API definition on our side (and you can view a preview and its future changelog entry), but the release step becomes manual. This means you can decide when to make an API definition live for your API users.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+When activated, uploading an API document of your API documentation will no longer automatically publish it. We continue to prepare this API definition on our side (and you can view a preview and its changelog), but the release step becomes manual. This means you can decide when to make an API definition live for your API users.
+
+This option allows you to continue working on your documentation without worrying about the impact on your users. For example, you can prepare a future definition of your API and its documentation with members of your organization and launch its publication when the time is right.
```
Here is a proposal, to make it a bit simpler to read. WDYT?
```suggestion
This option allows you to work on future versions of your API without disclosing the changes to your users until the moment you choose.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+#### Manually release a deployment
+
+From the list of your deployments, select an API document of your documentation with the "unreleased" status.
```
```suggestion
Select a deployment with the "unreleased" status from the list of your deployments.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+![](/images/help/deployments-list-not-released.png)
+
+The deployment preparation page will appear, allowing you to add context by adding a title or a brief description, for example, to accompany this new definition. These fields [support Markdown](/help/specification-support/markdown-support/) to make the result more readable.
```
```suggestion
The release preparation page will appear, allowing you to add context by adding a title or a brief description, for example, to explain the context of the change. These fields [support Markdown](/help/specification-support/markdown-support/) to make the result more readable.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+![](/images/help/deployment-release-form.png)
+
+### Unrelease an API definition
```
I'm not 100% sure of this one. It's more consistent with the previous title, but it doesn't 100% work. Please ignore it if you think it's better the way it's written now.
```suggestion
### Unrelease a deployment
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+### Unrelease an API definition
+
+It is possible to "unrelease" any previously released API definition of your documentation. By clicking on the definition, you will find the "Unrelease" option under the "..." button.
```
```suggestion
It is possible to "unrelease" any previously released deployment. By clicking on the deployment, you will find the "Unrelease" option under the "..." button.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+![](/images/help/unrelease-button.png)
+
+> Unreleasing a definition automatically adjusts your documentation: it will revert to the previous definition it was using, and its changelog will be adjusted accordingly to correctly reflect the changes.
```
```suggestion
> Unreleasing a deployment not only rolls back your API documentation to the previously released one. It also automatically reprocesses its changelog to correctly reflect the changes.
```

scharrier commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+
+> Unreleasing a definition automatically adjusts your documentation: it will revert to the previous definition it was using, and its changelog will be adjusted accordingly to correctly reflect the changes.
+{: .info}
+
+> We cannot restore a deleted API document.
+{: .warning}
```
I don't think we can "delete an API document" at the moment, what are you thinking about here?

jayk-u replied:
Maybe I'm wrong here but can't we delete an API document from the deployment list?
I've tried to remind here that if the user do so, we won't be able to restore the deleted document.

thimy replied:
Nope, we can only _unrelease_ it. We don't allow suppression so we have a clear history of what has been processed within a documentation.

scharrier commented on src/_help/publish-documentation/index.md:
```diff
@@ -0,0 +1,16 @@
+
+Bump.sh offers various tools to simplify this process.
+
+Deploying your API definition files can be done in several ways, through our website, API, or by integrating with your CI pipeline. You can choose between automatic deployments or manual validation to have more control over the published content or share work-in-progress variants with your team.
```
```suggestion
Deploying your API documents can be done in several ways, through our website, API, CLI, or by integrating with your CI pipeline. You can choose between automatic releases or manual validation to have more control over the published content or share work-in-progress variants with your team.
```

scharrier commented on src/_help/publish-documentation/index.md:
```diff
@@ -0,0 +1,16 @@
+
+Deploying your API definition files can be done in several ways, through our website, API, or by integrating with your CI pipeline. You can choose between automatic deployments or manual validation to have more control over the published content or share work-in-progress variants with your team.
+
+This section also covers the concept of branching, which allows you to have multiple versions of an API (such as v1.0/v2.0 or production and staging).
```
```suggestion
This section also covers the concept of branching, which allows you to have multiple versions and environments of an API (such as v1.0/v2.0 or production and staging).
```

scharrier commented on src/_help/publish-documentation/index.md:
```diff
@@ -0,0 +1,16 @@
+---
+title: Publish documentation
```
Shouldn't we add links to the different sections directly from this page?

scharrier reviewed (commented):
I did another round of comments/suggessiont. It's another big one, because of the needed clarification around the wording. I think it's now clearer, with the following concepts:
* API document: the file to upload
* API definition: the content of an API document (= where the API is described, also called API description)
* Deployment: the processing of an API document
* Release: making the result of a deployment visible to users (= updating doc + changelog entry)

I also tried to simplify some parts.
Let me know if you need another discussion about this whole part.

Thanks for your patience! I think that after this review/fixes we'll be able to merge this big one! 💪

thimy commented on src/_help/publish-documentation/documentation-access-management.md:
```diff
@@ -0,0 +1,52 @@
+
+Members of an organization have access to all the documentation contained within that organization.
+
+Inviting a colleague or teammate to become a member of [your organization](/help/organizations/) can be done from the settings of that organization. Simply enter the email address of the person in question.
```
```suggestion
Inviting a colleague or teammate to become a member of [your organization](/help/organizations/) can be done from the settings of that organization by simply entering their email address in the dedicated form.
```

jayk-u commented on src/_help/publish-documentation/deploy-and-release-management.md:
```diff
@@ -0,0 +1,85 @@
+- TOC
+{:toc}
+
+## What are deployments
```
I was not using the interrogative form here. I feel like keeping it that way.

jayk-u commented on src/_help/publish-documentation/branching.md:
```diff
@@ -0,0 +1,64 @@
+
+You can create as many branches as necessary, and each of these branches can receive its own deploys and has a distinct changelog from others. You can choose which branch will be displayed and used by default for deployments (when no target branch is designated).
+
+Switching from one API definition to another on documentation can be done in an instant through a dropdown menu.
```
Added part of your suggestion (but not the "public page" which has not been used anywhere else so far, could change later)

scharrier commented:
As this PR has become too complex, I am closing it in favor of a new clean one, which will present the latest version on this one + several fixes (UI ones + typos + redirections). It will be easier to finalize and merge it tomorrow.
